portal-kernel/src/com/liferay/document/library/kernel/util/DLPreviewableProcessor.java

```diff
@@ -99,6 +105,33 @@ public static void deleteFiles() {
 			});
 	}
 
+	public static void generatePDFPreviews(long userId) throws PortalException {
+		CompanyLocalServiceUtil.forEachCompanyId(
+			companyId -> {
+				try {
+					String jobName = "generatePDFPreviews-".concat(
+						String.valueOf(companyId));
```

Comment:
Why not `"generatePDFPreviews-" + companyId`? It looks much more simple.

Reply:
Also, isn't a bit overkill to add a PDF specific method to the DLPreviewableProcessor class? Can we add this to PDFProcessor instead, move this logic to PDFProcessorImpl and inject that component in the server admin MVCAction?

Reply:
done